src/CommandObserver.ts

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import * as Constants from './constants';
+
+const regex = /^(,)?(((?<origin>([^\s].*))\((?<linedetails>(\d+|\d+,\d+|\d+,\d+,\d+,\d+))\))|()):\s(?<subcategory>(()|([^:]*? )))(?<category>(error|warning))(\s*(?<code>[^: ]*))?\s*:\s(?<text>.*)$/gm;
```

Comment:
If ```origin``` matches everything except for spaces, shouldn't there be something after that capture group to match ```\s```? It seems like this will have to back-track a lot, as it will consume all characters first and then start to go backwards yielding them to ```linedetails``` group, no?

Reply:
Also maybe move this constant to wherever it's used, rather than in a global scope?

Reply:
Origin does match spaces. Yes it would have to backtrack a bit to get the line details. Do you think there is a better way to do this?

Reply:
Actually if I think about it, it would not have to backtrack. It tries to match everything until it gets a line delimiter and then matches the line details.